app/controllers/resources_controller.rb

```diff
@@ -38,8 +38,45 @@ def suggestions
     render json: ToolFilterService.new(params).call, include: params[:include], fields: field_params, status: :ok
   end
 
+  def publish_translation
+    if valid_publish_params?
+      render json: publish_translations, status: :ok
+    else
+      render json: {errors: {"errors" => [{source: {pointer: "/data/attributes/id"}, detail: "Record not found."}]}}, status: :unprocessable_entity
+    end
+  end
+
   private
 
+  def valid_publish_params?
+    params.dig("data", "relationships", "languages", "data") && params["resource_id"]
+  end
+
+  def publish_translations
+    draft_translations = []
+    languages = params["data"]["relationships"]["languages"]["data"]
+
+    languages.each do |lang|
+      draft_translations << publish_translation_for_language(lang)
+    end
+    draft_translations
+  end
+
+  def publish_translation_for_language(language_data)
+    draft_translation = find_or_create_latest_translation(language_data["id"])
+    if draft_translation
+      PublishTranslationJob.perform_async(draft_translation.id)
+      draft_translation
+    end
+  end
+
+  def find_or_create_latest_translation(language_id)
+    resource = Resource.find(params["resource_id"])
+
+    draft = resource.create_draft(language_id) if resource
```

Comment:
it looks like the `create_draft` method will always create a new draft even if another draft already exists for that resource & language.

Reply:
We want to re-use an existing draft when it exists, so that if there is an error publishing the draft we don't just keep creating new draft translations.

Reply:
ok, I'll revise it